samples/net/wifi/boards/xiao_esp32c3.conf

```diff
@@ -0,0 +1,11 @@
+CONFIG_WIFI=y
```

Comment:
@rabarar since the commit "title" is also too long, as per the compliance check report, I'd advice to just use the below verbatim, as this should solve both issues:

```
samples: net: wifi: add missing .conf and .overlay files for xiao_esp32c3

The conf and overlay files to properly compile the wifi samples were
omitted for the Xiao_esp32c3 board. I've added those two files and confirmed
that the sample properly compiles and runs.  

Signed-off-by: Robert Baruch <robertbaruch@mac.com>
```

I know this may sound overly pedantic, but with 10s of thousands of commit in the repo, we are trying to follow some basic rules to make digging into the history easier etc. And sign-off is also important to track code provenance and Developer Certificate of Origin. Thank you for your patience!